src/Microsoft.Bcl.HashCode/src/Microsoft.Bcl.HashCode.csproj

```
@@ -10,8 +10,8 @@
   <!-- Package servicing properties -->
   <PropertyGroup>
     <IsPackable>false</IsPackable>
-    <VersionPrefix>1.1.1</VersionPrefix>
-    <VersionPrefix Condition="'$(IsPackable)' == 'true'">1.2.0</VersionPrefix>
+    <VersionPrefix>6.0.0</VersionPrefix>
```

@ericstj I'm basically mimicking what we did for Unsafe.

But now I'm wondering if that was correct. I'm thinking of two options:

- Shouldn't this initial VersionPrefix value still be 1.1.1, so it matches PackageValidationBaselineVersion?
- Or alternatively, should PackageValidationBaselineVersion be changed to be 6.0.0?
-

---

> Shouldn't this initial VersionPrefix value still be 1.1.1, so it matches PackageValidationBaselineVersion?

Yeah.

>  should PackageValidationBaselineVersion be changed to be 6.0.0?

No, since that doesn't exist it wouldn't restore.

IMO we don't gain much having IsPackable turned off for everything.  Makes me wonder if we should be setting that right now since we want to ship all these packages.

---

> Makes me wonder if we should be setting that right now since we want to ship all these packages.

I think the idea of having everything turned off behind the IsPackable condition is that when we decide to turn it on, we just do it with one single property passed and we're done. Otherwise, we would have to revisit all the projects and add the condition again for the future releases. So better to just have it right now as is.

---

Right, but I'm saying our next release is meant to turn it on.  So why don't we build that way right now?  You can go apply `IsPackable` to everything you intend to ship.  Just as we would if we were getting ready to service an individual package, but do it for all.

My understanding of our current system is that `VersionPrefix` without the condition should represent the last shipped version - that way other packages that are building will get a dependency to the last shipped version.  Right now if we turned on only one other package that referenced BCL.HashCode it would get a dependency to version `6.0.0` which [doesn't exist.](https://www.nuget.org/packages/Microsoft.Bcl.HashCode/1.1.1#versions-body-tab)

I think you need `VersionPrefix` to be `1.1.1` with the next version as `6.0.0`.

---

Why update the package version to 6.0.0 for the next release?

---

To ensure it gets a higher file-version that what previously shipped.  Today file version and package version are tightly coupled in the arcade infra.  We benefit from the versioning logic in arcade as it ensures monotonically increasing versions that encode all relevant inputs.

To decouple these we'd need hacks here in m-p, or need to build new features into arcade.  The way this worked before in a decoupled way was due to pkgproj in corefx.  Seems to me like the smoother pathway here is to just satisfy the arcade rules.  We've also tossed around the idea of simplifying all the versions of m-p packages to a new major.minor - but that might be confusing to customers and send the wrong message for some of these legacy packages.

---

OK, that's unfortunate as the major version increment communicates that this package changed in a "major" way. But I'm OK with it if there isn't a straightforward way to decouple package version from everything else.

What Eric said above is true, `VersionPrefix` should point to the current version:

```suggestion
    <VersionPrefix>1.1.1</VersionPrefix>
```